examples/Vendored-XCFramework12-Example/BananaLib/build.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env sh
```

Comment:
any way this could be shared between this example and the existing one here? https://github.com/CocoaPods/CocoaPods/blob/master/examples/Vendored%20XCFramework%20Example/CoconutLib/build.sh? Not a blocker for this PR, but would be nice to not have to maintain both

Reply:
@amorde The scripts are similar, but there are a few differences between each and I worried a little about overcomplicating the setup by trying too hard to avoid the duplication.

The new Banana script that's compiling the Xcode 11 format xcframework is the closest to the existing script: both collect the dSYMs at the root and manually stuff the BCSymbolMaps down inside the .frameworks. The current differences are:
- Location of xcarchives (new in `build/DerivedData/`, old in `build/`)
- The new script does not have a need for static linking or library-style output. (I'm not as familiar with the setup for static linking, so I'm a little hesitant to refactor in a way that affects that portion of the existing script)

The new Coconut script is the Xcode 12 style compilation that removes the manual manipulation of debug symbols and instead uses the new `xcodebuild -create-xcframework -debug-symbols` option.

So, I think if we were to consolidate we'd need to consider:
- Where does the master copy of the script live where it can be called for both example projects?
- Fully generalize the script so it can be argument-controlled to pass in:
  - Source project/directory to compile
  - Output directory
  - Static/Dynamic Library/Framework output
  - Xcode 11 vs Xcode 12 debug symbol collection
- Maybe setup new wrapper build scripts that call into the master script so that there's less of a chance of passing in the wrong arguments when re-running.

You're definitely right that there's a good bit of similarity between parts of the script, I just fear that to properly setup a consolidated script that replicates what these do, we'd end up losing some of the more straightforward traceability that the current ones have. Let me know what you think!